src/sulogin-shell/sulogin-shell.c

```diff
@@ -41,6 +43,32 @@ static int reload_manager(sd_bus *bus) {
         return 0;
 }
 
+static int is_inactive_default_target(sd_bus *bus, bool *ret_is_inactive) {
```

Comment:
Not a big matter, but we usually name boolean arguments with positive meaning. So, please rename the function to `default_target_is_active()`.

Reply:
But what I'm interested in here is about inactiveness of default.target, not activeness. The function name indicates that explicitly. There are 7 states in the unit active states:

```c
typedef enum UnitActiveState {
        UNIT_ACTIVE,
        UNIT_RELOADING,
        UNIT_INACTIVE,
        UNIT_FAILED,
        UNIT_ACTIVATING,
        UNIT_DEACTIVATING,
        UNIT_MAINTENANCE,
        _UNIT_ACTIVE_STATE_MAX,
        _UNIT_ACTIVE_STATE_INVALID = -EINVAL,
} UnitActiveState;
```

The condition !active overlooks possibility of reloading, inactive, failed, activating, deactivating and maintenance, even in which case, systemctl default is tried unexpectedly.